cmd/antrea-agent/agent.go

```diff
@@ -267,8 +263,18 @@ func run(o *Options) error {
 	if err != nil {
 		return fmt.Errorf("error initializing agent: %v", err)
 	}
-	nodeConfig := agentInitializer.GetNodeConfig()
 
+	// Register Antrea Agent metrics if EnablePrometheusMetrics is set, and run a goroutine to register
+	// OVS meter metric if the OVS Meter feature is supported.
+	if *o.config.EnablePrometheusMetrics {
+		metrics.InitializePrometheusMetrics()
+		if openflow.OvsMetersAreSupported() {
+			ovsMeterStatsCollector := collector.NewOVSMeterStatsCollector(ofClient)
+			go ovsMeterStatsCollector.Run(stopCh)
+		}
```

Comment:
I feel perhaps we don't need a separate struct for this collector. It just passes the request to OpenFlow client anyway and we already spawn some goroutines in OpenFlow clients.
We could just pass `*o.config.EnablePrometheusMetrics` to OpenFlow client and start a goroutine when it's true, which should be more straightforward and expose less details to main.go.

We can add `Run(stopCh)` method to OpenFlow client and call `startPacketInHandler` and the metrics collector in it.

Reply:
Thanks, done. Please help check the latest version.